src/mscorlib/shared/System/Text/StringBuilder.Debug.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.Serialization;
+
+namespace System.Text
+{
+    public sealed partial class StringBuilder : ISerializable
+    {
+        private void ShowChunks(int maxChunksToShow = 10)
+        {
+            Debug.WriteLine('|' + string.Join('|', ShowChunksInOrder(maxChunksToShow)) + '|');
+        }
+
+        private IEnumerable<string> ShowChunksInOrder(int maxChunksToShow)
+        {
+            int numChunksToShow = 0;
+            StringBuilder lastChunk = null;
+            // Gets numChunksToShow. If numChunksToShow is larger than maxChunksToShow, then returns last chunk
+            GetNumChunksToShow(this, ref numChunksToShow, maxChunksToShow, ref lastChunk);
```

Comment:
ref int is unusual - if you do not recurse you can simply make it a return value.

consider renaming `lastChunk` to `headChunk` since last is a bit ambiguous (it's first logically, last in the datastructure)

rather than ref parameters, consider returning a tuple (int numChunksToShow, StringBuilder headChunk)